File: program33.java
//program to check whether number is perfect or not
import java.lang.*;
import java.util.*;

class Perfect
{
    int sum=0;

    public boolean CheckPerfect(int no)
    {
        if(no<0)
        {
            no=-no;
        }
        for(int i=1;i<=(no/2) &&(sum<no);i++)
        {
            if(no%i==0)
            {
                sum=sum+i;
            }
        }
        if(sum==no)
        {
            return true;
        }
        else
        {
            return false;
        }


    }
}

class program33
{
    public static void main(String arg[])
    {
        Perfect obj=new Perfect();
        int no=0;
        boolean ret=false;

        Scanner sobj=new Scanner(System.in);
        
        System.out.println("Enter nummber");
        no=sobj.nextInt();

        ret=obj.CheckPerfect(no);
        if(ret==true)
        {
            System.out.println("Number is perfect");
        }
        else
        {
            System.out.println("Number is not perfect");
        }


    }

}

File: program33.c
//check whether number is perfect or not
//when there is check whether in the problem them there
//if the sum of the factors of the number is equal to that number then it is perfect number
//should be boolean
#include<stdio.h>
#include<stdbool.h>
int SumOfFactors(int iNo)
{
    int iCnt=0,iSum=0;
    if(iNo<0)
    {
        iNo=-iNo;
    }
    for(iCnt=1;iCnt<=(iNo/2);iCnt++)
    {
        if(iNo%iCnt==0)
        {
            iSum=iSum+iCnt;
        }
    }
    return iSum;
}

bool CheckPerfect(int iNo)
{
    int iAns=0;
    iAns=SumOfFactors(iNo);
    if(iAns==iNo)
    {
        return true;
    }
    else
    {
        return false;
    }

}

int main()
{
    int iValue=0;
    bool bRet;

    printf("Enter number\n");
    scanf("%d",&iValue);

    bRet=CheckPerfect(iValue);
    if(bRet==true)
    {
        printf("%d is perfect number\n",iValue);
    }
    else
    {
        printf("%d is not perfect number\n",iValue);
    }
    return 0;
}

//perfect numbers
// 6  28  496